### PR TITLE
BSO Infernal Slayer Helm fix

### PR DESCRIPTION
**Pull request by TastyPumPum**
Fixes the infernal slayer helmet boosts for BSO
Also fixes the duplicate DHL/DHCB boost code
- [ ] I have tested all my changes thoroughly.

### CODE REVIEW

File: src/mahoji/lib/abstracted_commands/minionKill.ts [hunk 1]
```diff
@@ -369,18 +369,30 @@ export async function minionKillCommand(
...
+			: user.hasEquippedOrInBank('Infernal slayer helmet');
 		const hasBlackMask = monster?.canBePked
 			? wildyGear.hasEquipped('Black mask')
 			: user.hasEquippedOrInBank('Black mask');
-		const hasBlackMaskI = isOnTask
+		const hasBlackMaskI = monster?.canBePked
 			? wildyGear.hasEquipped('Black mask (i)')
 			: user.hasEquippedOrInBank('Black mask (i)');
 
 		if (attackStyles.includes(SkillsEnum.Ranged) || attackStyles.includes(SkillsEnum.Magic)) {
```
Comment (nwjgit) [inline src/mahoji/lib/abstracted_commands/minionKill.ts:385]: This doesn't take into account Infernal slayer helmet(i) while training melee, also L329 to L346 should be removed

Comment (TastyPumPum) [reply src/mahoji/lib/abstracted_commands/minionKill.ts:351]: Fixed I think?

Comment (nwjgit) [reply src/mahoji/lib/abstracted_commands/minionKill.ts:351]: You still want the checks for range and mage though, it should be something like this https://github.com/oldschoolgg/oldschoolbot/issues/5445#issuecomment-1773842000

Comment (nwjgit) [reply src/mahoji/lib/abstracted_commands/minionKill.ts:351]: I miss read your code, my bad. I believe this will work!

**Pull request merged**